tools/bifrost-make-tof-lookup-table.ipynb

```diff
@@ -0,0 +1,228 @@
+   "execution_count": null,
+   "id": "1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ess.reduce import time_of_flight\n",
+    "import sciline\n",
+    "import scipp as sc\n",
+    "from scippneutron.chopper import DiskChopper\n",
+    "from ess.spectroscopy.indirect.workflow import load_everything\n",
+    "from ess.bifrost.data import simulated_elastic_incoherent_with_phonon"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2",
+   "metadata": {},
+   "source": [
+    "## Load data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_filename = simulated_elastic_incoherent_with_phonon()\n",
+    "sample, triplets, _, raw_choppers, monitors, _ = load_everything(input_filename)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "source_position = next(iter(triplets.values())).coords['source_position']\n",
+    "sample_position = sample['position']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5",
+   "metadata": {},
+   "source": [
+    "## Process choppers"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def parse_choppers(raw_choppers, source_position):\n",
+    "    choppers = sc.DataGroup()\n",
+    "    for name, chopper in raw_choppers.items():\n",
+    "        processed = chopper.copy()\n",
+    "        # These are constant in the simulated data.\n",
+    "        processed['rotation_speed'] = processed['rotation_speed'].data.mean()\n",
+    "        processed['phase'] = processed['phase'].data.mean()\n",
+    "        # Guessing here as this is not stored in the file.\n",
+    "        processed['beam_position'] = sc.scalar(0.0, unit='deg')\n",
+    "        # ess.reduce.time_of_flight.simulate_beamline requires the\n",
+    "        # positions to be relative to the source.\n",
+    "        processed['position'] = chopper['position'] - source_position\n",
+    "        choppers[name] = DiskChopper.from_nexus(processed)\n",
+    "    return choppers"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "choppers = parse_choppers(raw_choppers, source_position)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8",
+   "metadata": {},
+   "source": [
+    "## Setting up the workflow"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9",
+   "metadata": {},
+   "source": [
+    "The table needs to cover the range of distances for both sample and monitor.\n",
+    "Here, the shifts of 0.1m are arbitrarily chosen to make sure the table covers `l1` and `l_monitor`.\n",
+    "We could probably save space by making 2 smaller tables for narrow ranges around `l1` and `l_monitor`, respectively."
```

Comment:
In my experience, the distance resolution in the table does not seem to matter that much.
The number of neutrons used however seems to matter a lot.
I am not sure it would be worth making 2 tables.